// decap-cms-widget-markdown/dist/esm/components/plate-ui/command.js
"use strict";
'use client';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Command = void 0;
exports.CommandDialog = CommandDialog;
exports.CommandShortcut = exports.CommandSeparator = exports.CommandList = exports.CommandItem = exports.CommandInput = exports.CommandGroup = exports.CommandEmpty = void 0;
var React = _interopRequireWildcard(require("react"));
var _cn = require("@udecode/cn");
var _cmdk = require("cmdk");
var _icons = require("@/components/icons");
var _dialog = require("./dialog");
var _react2 = require("@emotion/react");
const _excluded = ["children"],
  _excluded2 = ["className"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
const Command = exports.Command = (0, _cn.withCn)(_cmdk.Command, 'flex h-full w-full flex-col overflow-hidden rounded-md bg-popover text-popover-foreground');
function CommandDialog(_ref) {
  let {
      children
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  return (0, _react2.jsx)(_dialog.Dialog, props, (0, _react2.jsx)(_dialog.DialogContent, {
    className: "overflow-hidden p-0 shadow-lg"
  }, (0, _react2.jsx)(Command, {
    className: "[&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-muted-foreground [&_[cmdk-group]:not([hidden])_~[cmdk-group]]:pt-0 [&_[cmdk-group]]:px-2 [&_[cmdk-input-wrapper]_svg]:h-5 [&_[cmdk-input-wrapper]_svg]:w-5 [&_[cmdk-input]]:h-12 [&_[cmdk-item]]:px-2 [&_[cmdk-item]]:py-3 [&_[cmdk-item]_svg]:h-5 [&_[cmdk-item]_svg]:w-5"
  }, children)));
}
const CommandInput = exports.CommandInput = (0, _cn.withRef)((_ref2, ref) => {
  let {
      className
    } = _ref2,
    props = _objectWithoutProperties(_ref2, _excluded2);
  return (0, _react2.jsx)("div", {
    className: "flex items-center border-b px-3",
    "cmdk-input-wrapper": ""
  }, (0, _react2.jsx)(_icons.Icons.search, {
    className: "mr-2 h-4 w-4 shrink-0 opacity-50"
  }), (0, _react2.jsx)(_cmdk.Command.Input, _extends({
    ref: ref,
    className: (0, _cn.cn)('flex h-11 w-full rounded-md bg-transparent py-3 text-sm outline-none placeholder:text-muted-foreground disabled:cursor-not-allowed disabled:opacity-50', className)
  }, props)));
});
const CommandList = exports.CommandList = (0, _cn.withCn)(_cmdk.Command.List, 'max-h-[500px] overflow-y-auto overflow-x-hidden');
const CommandEmpty = exports.CommandEmpty = (0, _cn.withCn)(_cmdk.Command.Empty, 'py-6 text-center text-sm');
const CommandGroup = exports.CommandGroup = (0, _cn.withCn)(_cmdk.Command.Group, 'overflow-hidden p-1 text-foreground [&_[cmdk-group-heading]]:px-2 [&_[cmdk-group-heading]]:py-1.5 [&_[cmdk-group-heading]]:text-xs [&_[cmdk-group-heading]]:font-medium [&_[cmdk-group-heading]]:text-muted-foreground');
const CommandSeparator = exports.CommandSeparator = (0, _cn.withCn)(_cmdk.Command.Separator, '-mx-1 h-px bg-border');
const CommandItem = exports.CommandItem = (0, _cn.withCn)(_cmdk.Command.Item, 'relative flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none aria-selected:bg-accent aria-selected:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50');
const CommandShortcut = exports.CommandShortcut = (0, _cn.withCn)((0, _cn.createPrimitiveElement)('span'), 'ml-auto text-xs tracking-widest text-muted-foreground');
