// decap-cms-widget-markdown/dist/esm/components/plate-ui/dropdown-menu.js
"use strict";
'use client';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useOpenState = exports.DropdownMenuTrigger = exports.DropdownMenuSubTrigger = exports.DropdownMenuSubContent = exports.DropdownMenuSub = exports.DropdownMenuShortcut = exports.DropdownMenuSeparator = exports.DropdownMenuRadioItem = exports.DropdownMenuRadioGroup = exports.DropdownMenuPortal = exports.DropdownMenuLabel = exports.DropdownMenuItem = exports.DropdownMenuGroup = exports.DropdownMenuContent = exports.DropdownMenuCheckboxItem = exports.DropdownMenu = void 0;
var _react = _interopRequireWildcard(require("react"));
var React = _react;
var DropdownMenuPrimitive = _interopRequireWildcard(require("@radix-ui/react-dropdown-menu"));
var _cn = require("@udecode/cn");
var _classVarianceAuthority = require("class-variance-authority");
var _icons = require("@/components/icons");
var _react2 = require("@emotion/react");
const _excluded = ["className", "inset", "children"],
  _excluded2 = ["className", "children"],
  _excluded3 = ["className", "children", "hideIcon"];
function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function (e) { return e ? t : r; })(e); }
function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && Object.prototype.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
function _objectDestructuringEmpty(obj) { if (obj == null) throw new TypeError("Cannot destructure " + obj); }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
const DropdownMenu = exports.DropdownMenu = DropdownMenuPrimitive.Root;
const DropdownMenuTrigger = exports.DropdownMenuTrigger = DropdownMenuPrimitive.Trigger;
const DropdownMenuGroup = exports.DropdownMenuGroup = DropdownMenuPrimitive.Group;
const DropdownMenuPortal = exports.DropdownMenuPortal = DropdownMenuPrimitive.Portal;
const DropdownMenuSub = exports.DropdownMenuSub = DropdownMenuPrimitive.Sub;
const DropdownMenuRadioGroup = exports.DropdownMenuRadioGroup = DropdownMenuPrimitive.RadioGroup;
const DropdownMenuSubTrigger = exports.DropdownMenuSubTrigger = (0, _cn.withRef)((_ref, ref) => {
  let {
      className,
      inset,
      children
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  return (0, _react2.jsx)(DropdownMenuPrimitive.SubTrigger, _extends({
    ref: ref,
    className: (0, _cn.cn)('flex cursor-default select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none focus:bg-accent data-[state=open]:bg-accent', 'data-[disabled]:pointer-events-none data-[disabled]:opacity-50', inset && 'pl-8', className)
  }, props), children, (0, _react2.jsx)(_icons.Icons.chevronRight, {
    className: "ml-auto h-4 w-4"
  }));
});
const DropdownMenuSubContent = exports.DropdownMenuSubContent = (0, _cn.withCn)(DropdownMenuPrimitive.SubContent, 'z-50 min-w-[8rem] overflow-hidden rounded-md border bg-popover p-1 text-popover-foreground shadow-lg data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2');
const DropdownMenuContentVariants = (0, _cn.withProps)(DropdownMenuPrimitive.Content, {
  sideOffset: 4,
  className: (0, _cn.cn)('z-50 min-w-[8rem] overflow-hidden rounded-md border bg-popover p-1 text-popover-foreground shadow-md data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2')
});
const DropdownMenuContent = exports.DropdownMenuContent = (0, _cn.withRef)((_ref2, ref) => {
  let props = Object.assign({}, (_objectDestructuringEmpty(_ref2), _ref2));
  return (0, _react2.jsx)(DropdownMenuPrimitive.Portal, null, (0, _react2.jsx)(DropdownMenuContentVariants, _extends({
    ref: ref
  }, props)));
});
const menuItemVariants = (0, _classVarianceAuthority.cva)((0, _cn.cn)('relative flex h-9 cursor-pointer select-none items-center rounded-sm px-2 py-1.5 text-sm outline-none transition-colors', 'focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50'), {
  variants: {
    inset: {
      true: 'pl-8'
    }
  }
});
const DropdownMenuItem = exports.DropdownMenuItem = (0, _cn.withVariants)(DropdownMenuPrimitive.Item, menuItemVariants, ['inset']);
const DropdownMenuCheckboxItem = exports.DropdownMenuCheckboxItem = (0, _cn.withRef)((_ref3, ref) => {
  let {
      className,
      children
    } = _ref3,
    props = _objectWithoutProperties(_ref3, _excluded2);
  return (0, _react2.jsx)(DropdownMenuPrimitive.CheckboxItem, _extends({
    ref: ref,
    className: (0, _cn.cn)('relative flex select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50', 'cursor-pointer', className)
  }, props), (0, _react2.jsx)("span", {
    className: "absolute left-2 flex h-3.5 w-3.5 items-center justify-center"
  }, (0, _react2.jsx)(DropdownMenuPrimitive.ItemIndicator, null, (0, _react2.jsx)(_icons.Icons.check, {
    className: "h-4 w-4"
  }))), children);
});
const DropdownMenuRadioItem = exports.DropdownMenuRadioItem = (0, _cn.withRef)((_ref4, ref) => {
  let {
      className,
      children,
      hideIcon
    } = _ref4,
    props = _objectWithoutProperties(_ref4, _excluded3);
  return (0, _react2.jsx)(DropdownMenuPrimitive.RadioItem, _extends({
    ref: ref,
    className: (0, _cn.cn)('relative flex select-none items-center rounded-sm pl-8 pr-2 text-sm outline-none transition-colors focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50', 'h-9 cursor-pointer px-2 data-[state=checked]:bg-accent data-[state=checked]:text-accent-foreground', className)
  }, props), !hideIcon && (0, _react2.jsx)("span", {
    className: "absolute right-2 flex h-3.5 w-3.5 items-center justify-center"
  }, (0, _react2.jsx)(DropdownMenuPrimitive.ItemIndicator, null, (0, _react2.jsx)(_icons.Icons.check, {
    className: "h-4 w-4"
  }))), children);
});
const dropdownMenuLabelVariants = (0, _classVarianceAuthority.cva)((0, _cn.cn)('select-none px-2 py-1.5 text-sm font-semibold'), {
  variants: {
    inset: {
      true: 'pl-8'
    }
  }
});
const DropdownMenuLabel = exports.DropdownMenuLabel = (0, _cn.withVariants)(DropdownMenuPrimitive.Label, dropdownMenuLabelVariants, ['inset']);
const DropdownMenuSeparator = exports.DropdownMenuSeparator = (0, _cn.withCn)(DropdownMenuPrimitive.Separator, '-mx-1 my-1 h-px bg-muted');
const DropdownMenuShortcut = exports.DropdownMenuShortcut = (0, _cn.withCn)((0, _cn.createPrimitiveElement)('span'), 'ml-auto text-xs tracking-widest opacity-60');
const useOpenState = () => {
  const [open, setOpen] = (0, _react.useState)(false);
  const onOpenChange = (0, _react.useCallback)((_value = !open) => {
    setOpen(_value);
  }, [open]);
  return {
    open,
    onOpenChange
  };
};
exports.useOpenState = useOpenState;
