// decap-cms-widget-markdown/dist/esm/components/plate-ui/fixed-toolbar-buttons.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.FixedToolbarButtons = FixedToolbarButtons;
var _react = _interopRequireDefault(require("react"));
var _plateBasicMarks = require("@udecode/plate-basic-marks");
var _plateCommon = require("@udecode/plate-common");
var _icons = require("@/components/icons");
var _insertDropdownMenu = require("./insert-dropdown-menu");
var _markToolbarButton = require("./mark-toolbar-button");
var _modeDropdownMenu = require("./mode-dropdown-menu");
var _toolbar = require("./toolbar");
var _turnIntoDropdownMenu = require("./turn-into-dropdown-menu");
var _react2 = require("@emotion/react");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function FixedToolbarButtons() {
  const readOnly = (0, _plateCommon.useEditorReadOnly)();
  return (0, _react2.jsx)("div", {
    className: "w-full overflow-hidden"
  }, (0, _react2.jsx)("div", {
    className: "flex flex-wrap",
    style: {
      transform: 'translateX(calc(-1px))'
    }
  }, !readOnly && (0, _react2.jsx)(_react.default.Fragment, null, (0, _react2.jsx)(_toolbar.ToolbarGroup, {
    noSeparator: true
  }, (0, _react2.jsx)(_insertDropdownMenu.InsertDropdownMenu, null), (0, _react2.jsx)(_turnIntoDropdownMenu.TurnIntoDropdownMenu, null)), (0, _react2.jsx)(_toolbar.ToolbarGroup, null, (0, _react2.jsx)(_markToolbarButton.MarkToolbarButton, {
    tooltip: "Bold (\u2318+B)",
    nodeType: _plateBasicMarks.MARK_BOLD
  }, (0, _react2.jsx)(_icons.Icons.bold, null)), (0, _react2.jsx)(_markToolbarButton.MarkToolbarButton, {
    tooltip: "Italic (\u2318+I)",
    nodeType: _plateBasicMarks.MARK_ITALIC
  }, (0, _react2.jsx)(_icons.Icons.italic, null)), (0, _react2.jsx)(_markToolbarButton.MarkToolbarButton, {
    tooltip: "Underline (\u2318+U)",
    nodeType: _plateBasicMarks.MARK_UNDERLINE
  }, (0, _react2.jsx)(_icons.Icons.underline, null)), (0, _react2.jsx)(_markToolbarButton.MarkToolbarButton, {
    tooltip: "Strikethrough (\u2318+\u21E7+M)",
    nodeType: _plateBasicMarks.MARK_STRIKETHROUGH
  }, (0, _react2.jsx)(_icons.Icons.strikethrough, null)), (0, _react2.jsx)(_markToolbarButton.MarkToolbarButton, {
    tooltip: "Code (\u2318+E)",
    nodeType: _plateBasicMarks.MARK_CODE
  }, (0, _react2.jsx)(_icons.Icons.code, null)))), (0, _react2.jsx)("div", {
    className: "grow"
  }), (0, _react2.jsx)(_toolbar.ToolbarGroup, {
    noSeparator: true
  }, (0, _react2.jsx)(_modeDropdownMenu.ModeDropdownMenu, null))));
}
