// decap-cms-widget-markdown/dist/esm/components/plate-ui/heading-element.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HeadingElement = void 0;
var _react = _interopRequireDefault(require("react"));
var _cn = require("@udecode/cn");
var _plateCommon = require("@udecode/plate-common");
var _classVarianceAuthority = require("class-variance-authority");
var _react2 = require("@emotion/react");
const _excluded = ["variant", "isFirstBlock", "children"];
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }
function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }
const headingVariants = (0, _classVarianceAuthority.cva)('', {
  variants: {
    variant: {
      h1: 'mb-1 mt-[2em] font-heading text-4xl font-bold',
      h2: 'mb-px mt-[1.4em] font-heading text-2xl font-semibold tracking-tight',
      h3: 'mb-px mt-[1em] font-heading text-xl font-semibold tracking-tight',
      h4: 'mt-[0.75em] font-heading text-lg font-semibold tracking-tight',
      h5: 'mt-[0.75em] text-lg font-semibold tracking-tight',
      h6: 'mt-[0.75em] text-base font-semibold tracking-tight'
    },
    isFirstBlock: {
      true: 'mt-0',
      false: ''
    }
  }
});
const HeadingElementVariants = (0, _cn.withVariants)(_plateCommon.PlateElement, headingVariants, ['isFirstBlock', 'variant']);
const HeadingElement = exports.HeadingElement = (0, _cn.withRef)((_ref, ref) => {
  let {
      variant = 'h1',
      isFirstBlock,
      children
    } = _ref,
    props = _objectWithoutProperties(_ref, _excluded);
  const {
    element,
    editor
  } = props;
  const Element = variant;
  return (0, _react2.jsx)(HeadingElementVariants, _extends({
    ref: ref,
    asChild: true,
    variant: variant,
    isFirstBlock: element === editor.children[0]
  }, props), (0, _react2.jsx)(Element, null, children));
});
