// decap-cms-widget-markdown/dist/esm/components/plate-ui/more-dropdown-menu.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MoreDropdownMenu = MoreDropdownMenu;
var _react = _interopRequireDefault(require("react"));
var _plateBasicMarks = require("@udecode/plate-basic-marks");
var _plateCommon = require("@udecode/plate-common");
var _icons = require("@/components/icons");
var _dropdownMenu = require("./dropdown-menu");
var _toolbar = require("./toolbar");
var _react2 = require("@emotion/react");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
function _extends() { _extends = Object.assign ? Object.assign.bind() : function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }
function MoreDropdownMenu(props) {
  const editor = (0, _plateCommon.useEditorRef)();
  const openState = (0, _dropdownMenu.useOpenState)();
  return (0, _react2.jsx)(_dropdownMenu.DropdownMenu, _extends({
    modal: false
  }, openState, props), (0, _react2.jsx)(_dropdownMenu.DropdownMenuTrigger, {
    asChild: true
  }, (0, _react2.jsx)(_toolbar.ToolbarButton, {
    pressed: openState.open,
    tooltip: "Insert"
  }, (0, _react2.jsx)(_icons.Icons.more, null))), (0, _react2.jsx)(_dropdownMenu.DropdownMenuContent, {
    align: "start",
    className: "flex max-h-[500px] min-w-[180px] flex-col gap-0.5 overflow-y-auto"
  }, (0, _react2.jsx)(_dropdownMenu.DropdownMenuItem, {
    onSelect: () => {
      (0, _plateCommon.toggleMark)(editor, {
        key: _plateBasicMarks.MARK_SUBSCRIPT,
        clear: _plateBasicMarks.MARK_SUPERSCRIPT
      });
      (0, _plateCommon.focusEditor)(editor);
    }
  }, (0, _react2.jsx)(_icons.Icons.superscript, {
    className: "mr-2 h-5 w-5"
  }), "Superscript"), (0, _react2.jsx)(_dropdownMenu.DropdownMenuItem, {
    onSelect: () => {
      (0, _plateCommon.toggleMark)(editor, {
        key: _plateBasicMarks.MARK_SUPERSCRIPT,
        clear: _plateBasicMarks.MARK_SUBSCRIPT
      });
      (0, _plateCommon.focusEditor)(editor);
    }
  }, (0, _react2.jsx)(_icons.Icons.subscript, {
    className: "mr-2 h-5 w-5"
  }), "Subscript")));
}
